enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,566 @@
+the controller will trigger a snapshot operation by creating a `volumeSnapshotConent` object, which will be 
+detected by the _CSI external-snapshotter sidecar_.
+
+MicroShift will embed the controller's manifests and deploy them during startup.  They are: 
+
+1. CSI VolumeSnapshot Controller Deployment
+2. CSI VolumeSnapshot Controller Service Account
+3. CSI VolumeSnapshot Controller ClusterRole and ClusterRoleBinding
+4. CSI VolumeSnapshot Controller Role and RoleBinding
+
+### CSI-External
+
+The CSI Snapshot Validation Webhook is maintained SIG-Storage upstream. Downstream OCP releases of this container 
+image will be used by MicroShift. The container image is specified by LVMS and is included in the _topolvm-controller_ 
+deployment. It watches for `volumeSnapshotContent` and `volumeSnapshotClass` events.  When an event is detected, it 
+makes the appropriate CSI gRPC call to LVMS via a shared unix socket.
+
+### CSI-Validation-Webhook
+
+The CSI Snapshot Validation Webhook is maintained SIG-Storage upstream. Downstream OCP releases of this container 
+image will be used by MicroShift.  The webhook serves as gatekeeper to `volumeSnapshot` CREATE and UPDATE events. 
+For the specifics of validation, _see_ 
+[Kubernetes KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook#validating-scenarios).
+
+1. Validating Webhook Deployment
+2. Validating Webhook Service
+3. ValidatingWebhookConfiguration
+4. Validating Webhook ClusterRole and ClusterRoleBinding
+
+### MicroShift Assets
+
+MicroShift manages the deployment of embedded component assets through an interface called the Service Manager. The 
+logic behind this interface deploys components in an order that respects interdependencies, waits for services to 
+start, and intelligently stops services on interrupt.  MicroShift's default CSI storage service is already managed 
+by the service manager. 
+
+MicroShift's CSI service manager will deploy the CSI Volume Snapshot components.  The files will be stored under the
+`microshift/assets/components/csi-external-snapshot-controller` directory.  The following files will be added: 
+
+1. `csi_controller_deployment.yaml` specifies the CSI Snapshot Controller
+2. `serviceaccount.yaml` specifies the controller's service account
+3. `05_operand_rbac.yaml` provides RBAC configuration to the CSI controller 
+4. `webhook_config.yaml`
+5. `webhook_deployment.yaml`
+6. `volumesnapshotclasses_snapshot.storage.k8s.io.yaml` defines the VolumeSnapshotClass CRD 
+7. `volumesnapshotcontents_snapshot.storage.k8s.io.yaml` defines the VolumeSnapshotContents CRD
+8. `volumesnapshots_snapshot.storage.k8s.io.yaml` defines the VolumeSnapshot CRD
+
+### Greenboot Changes
+
+MicroShift's greenboot scripts will be changed to include checks for the CSI Snapshot Controller and 
+WebhookValidation Pod.
```

Comment:
Let's add specifics here. Changing the "expected" pod namespace and count is done in the following place.
https://github.com/openshift/microshift/blob/809c1b9182aa04a9d40ac101ed60cb0d7b6a9d09/packaging/greenboot/microshift-running-check.sh#L6